### ISSUE TITLE
HangingBreakByEntityEvent, EntityInteractEvent, PlayerInteractEvent and PlayerArmorStandManipulateEvent

### ISSUE THREAD

**Issue by CarloDrift** [SkriptLang/Skript#3115]
### Description
These events are essential to be able to protect an area such as a lobby.

**Comment by FranKusmiruk** [comment]
HangingBreakByEntityEvent is handled by the `break` event, PlayerInteractEvent is handled by the `click` event. The rest are valid though.

**Comment by Eole7** [comment]
PlayerInteractEvent is not the same as the on click event, as it doesn't handle the case of an physical interaction (eg crop trample/trip wire/). It would be nice to have an event to handle this type of interactions

**Comment by CarloDrift** [comment]
@FranKusmiruk

**Comment by KingAlterIV** [comment]
PlayerInteractEvent is indeed the same event has the click event. There are action enums for it, and PHYSICAL is for what you said, crop trampling, pressure plates, redstone ore, etc.

![Action enums](https://i.imgur.com/H4Qdicc.png)

Skript already have a pressure plate event as well, and it uses the PHYSICAL enum with the PlayerInteractEvent.

![Pressure plate](https://i.imgur.com/STnhg92.png)

**Comment by Eole7** [comment]
> PlayerInteractEvent is indeed the same event has the click event.
But as I said, it is not triggered for PHYSICAL actions, so there is no way to detect these interactions except for pressure plate/tripwire where Skript has specific events for these.

I think that a more global event (eg "on physical interaction", with an optional parameter for pressure plates/crops/... ? ) would be preferable, so that we don't have to put 5 events for a “real” single event which is PlayerInteractEvent

**Comment by KingAlterIV** [comment]
Can be detected with skript-mirror.
https://i.imgur.com/uVabRKW.gifv

**Comment by Eole7** [comment]
Of course, but what does this change? If there is a suggestion posted here, it's to add it to Skript. 
And then if we follow this principle, since skript-mirror allows access to the whole Spigot API, why not simply stopping Skript development?

**Comment by FranKusmiruk** [comment]
> I think that a more global event ("on physical interaction", with an optional parameter for pressure plates/crops/... ? ) would be preferable, so that we don't have to put 5 events for a “real” single event which is PlayerInteractEvent

I'd rather have separate events for the separate actions that interaction type is triggered for, an event like the one being proposed is not intuitive at all and can lead to errors pretty easily. We should definitely support more triggers for this interaction though.



> @FranKusmiruk

The project developers get notifications for each issue/PR and its comments, the mention does nothing in this case.

**Comment by KingAlterIV** [comment]
> If there is a suggestion posted here, it's to add it to Skript.

skript-mirror's main usage is for stuff that aren't inside of Skript. Better off using it as an alternative for now, than waiting for Skript to implement it.  

> since skript-mirror allows access to the whole Spigot API, why not simply stopping Skript development?

skript-mirror can't fully access the whole Spigot API, as it still has limits.

**Comment by Olyno** [comment]
> skript-mirror's main usage is for stuff that aren't inside of Skript. Better off using it as an alternative for now, than waiting for Skript to implement it.

Is it really usefull to talk about skript-mirror here? We're on the Skript's repository, not the skript-mirror's repository. So please think about how implement this feature instead of find another way to do that.

**Comment by legofreak1999** [comment]
+ due to skript always having a higher event priority than mirror the cancelation skript does makes it impossible to make it with mirror.
